private_sharing/api_views.py

```diff
@@ -301,6 +330,10 @@ class ProjectFileUploadView(ProjectFormBaseView):
     def post(self, request):
         super().post(request)
 
+        # Check datatypes
+        if not self.good_datatypes:
```

Comment:
(this feels out of place since normally it'd be raised as an error in the form validation)

Reply:
See above about having to pass objects to the form